conjure_oxide/src/ast.rs

```diff
@@ -1,12 +1,18 @@
 use std::collections::HashMap;
 
-#[derive(Debug)]
+#[derive(Debug, PartialEq)]
 pub struct Model {
     pub variables: HashMap<Name, DecisionVariable>,
     pub constraints: Vec<Expression>,
 }
 
 impl Model {
+    pub fn new() -> Model {
+        Model {
+            variables: HashMap::new(),
+            constraints: Vec::new(),
+        }
+    }
```

Comment:
Do we want Model to be constructable by hand from the outside?

If not, might be good to do something like the following, which would restrict manual construction.

```suggestion
pub struct Model {
    pub variables: HashMap<Name, DecisionVariable>,
    pub constraints: Vec<Expression>,
    _private: ()
}

impl Model {
    pub fn new() -> Model {
        Model {
            variables: HashMap::new(),
            constraints: Vec::new(),
        }
    }
```

(structs with any private members are not constructible outside their module)

Reply:
This makes sense, but I would (for now) like them to be constructable as it allows for testing.

Reply:
Fair enough, was just not sure if you were aware of this or not.

I think unit tests (i.e. in file ones) can access private stuff from the module they are in - this might be a good option long term so that when private fields are inevitably added, your tests don't break.